Index: include/pthread.h
===================================================================
--- include/pthread.h	(revision 6329)
+++ include/pthread.h	(working copy)
@@ -265,11 +265,11 @@
 int WINPTHREAD_API pthread_setschedparam(pthread_t thread, int pol, const struct sched_param *param);
 
 /* synchronization objects */
-typedef void	*pthread_spinlock_t;
-typedef void	*pthread_mutex_t;
-typedef void	*pthread_cond_t;
-typedef void	*pthread_rwlock_t;
-typedef void	*pthread_barrier_t;
+typedef struct spin_t	*pthread_spinlock_t;
+typedef struct mutex_t	*pthread_mutex_t;
+typedef struct cond_t	*pthread_cond_t;
+typedef struct rwlock_t	*pthread_rwlock_t;
+typedef struct barrier_t	*pthread_barrier_t;
 
 #define PTHREAD_MUTEX_NORMAL 0
 #define PTHREAD_MUTEX_ERRORCHECK 1
@@ -279,14 +279,14 @@
 #define GENERIC_ERRORCHECK_INITIALIZER			((void *) (size_t) -2)
 #define GENERIC_RECURSIVE_INITIALIZER			((void *) (size_t) -3)
 #define GENERIC_NORMAL_INITIALIZER			((void *) (size_t) -1)
-#define PTHREAD_MUTEX_INITIALIZER			(pthread_mutex_t *)GENERIC_INITIALIZER
-#define PTHREAD_RECURSIVE_MUTEX_INITIALIZER		(pthread_mutex_t *)GENERIC_RECURSIVE_INITIALIZER
-#define PTHREAD_ERRORCHECK_MUTEX_INITIALIZER		(pthread_mutex_t *)GENERIC_ERRORCHECK_INITIALIZER
-#define PTHREAD_NORMAL_MUTEX_INITIALIZER		(pthread_mutex_t *)GENERIC_NORMAL_INITIALIZER
+#define PTHREAD_MUTEX_INITIALIZER			(pthread_mutex_t)GENERIC_INITIALIZER
+#define PTHREAD_RECURSIVE_MUTEX_INITIALIZER		(pthread_mutex_t)GENERIC_RECURSIVE_INITIALIZER
+#define PTHREAD_ERRORCHECK_MUTEX_INITIALIZER		(pthread_mutex_t)GENERIC_ERRORCHECK_INITIALIZER
+#define PTHREAD_NORMAL_MUTEX_INITIALIZER		(pthread_mutex_t)GENERIC_NORMAL_INITIALIZER
 #define PTHREAD_DEFAULT_MUTEX_INITIALIZER		PTHREAD_NORMAL_MUTEX_INITIALIZER
-#define PTHREAD_COND_INITIALIZER			(pthread_cond_t *)GENERIC_INITIALIZER
-#define PTHREAD_RWLOCK_INITIALIZER			(pthread_rwlock_t *)GENERIC_INITIALIZER
-#define PTHREAD_SPINLOCK_INITIALIZER			(pthread_spinlock_t *)GENERIC_INITIALIZER
+#define PTHREAD_COND_INITIALIZER			(pthread_cond_t)GENERIC_INITIALIZER
+#define PTHREAD_RWLOCK_INITIALIZER			(pthread_rwlock_t)GENERIC_INITIALIZER
+#define PTHREAD_SPINLOCK_INITIALIZER			(pthread_spinlock_t)GENERIC_INITIALIZER
 
 extern void WINPTHREAD_API (**_pthread_key_dest)(void *);
 int         WINPTHREAD_API pthread_key_create(pthread_key_t *key, void (* dest)(void *));
